pymc/aesaraf.py

```diff
@@ -564,15 +605,13 @@ def join_nonshared_inputs(
         tensor_type = joined.type
         inarray = tensor_type("inarray")
     else:
-        if point is None:
-            raise ValueError("A point is required when `make_shared` is True")
         joined_values = np.concatenate([point[var.name].ravel() for var in vars])
         inarray = aesara.shared(joined_values, "inarray")
 
     if aesara.config.compute_test_value != "off":
         inarray.tag.test_value = joined.tag.test_value
 
-    replace = {}
+    replace: Dict[TensorVariable, TensorSharedVariable] = {}
```

Comment:
This is incorrect, should be:
```suggestion
    replace: Dict[TensorVariable, TensorVariable] = {}
```

Reply:
Wouldn't it be the Union of TensorVariable and TensorSharedVariable? I am thinking when the `make_shared` is True where the `inarray` sections would be based off of `aesara.shared(joined_values, "inarray")`.

Reply:
When `make_shared` is True the replacement is still TensorVariable to TensorVariable. You are replacing the old variables by a slice of the joined input which is shared, but the slice itself is just another TensorVariable again.

Reply:
Cool. Thanks for the closer look

Reply:
Since some of the key, values here are from the `shared` argument, won't it be the union of TensorVariable and TensorSharedVariable? I think that was my original rational